src/orion/algo/dehb/dehb.py

```diff
@@ -0,0 +1,513 @@
+
+remove_loguru()
+
+from dehb.optimizers import DEHB as DEHBImpl
+from sspace.convert import convert_space
+from sspace.convert import transform as to_orion
+
+from orion.algo.base import BaseAlgorithm
+from orion.algo.space import Space
+from orion.core.utils import format_trials
+from orion.core.worker.trial import Trial
+
+logger = logging.getLogger(__name__)
+
+
+class UnsupportedConfiguration(Exception):
+    """Raised when an unsupported configuration is sent"""
+
+    pass
+
+
+SPACE_ERROR = """
+DEHB cannot be used if space does not contain a fidelity dimension.
+"""
+
+MUTATION_STRATEGIES = [
+    "rand1",
+    "rand2dir",
+    "randtobest1",
+    "currenttobest1",
+    "best1",
+    "best2",
+    "rand2",
+]
+
+CROSSOVER_STRATEGY = [
+    "bin",
+    "exp",
+]
+
+FIX_MODES = ["random", "clip"]
+
+
+# pylint: disable=too-many-public-methods
+class DEHB(DEHBImpl, BaseAlgorithm):
+    """Differential Evolution with HyperBand
+
+    This class is a wrapper around the librairy DEHB:
+    https://github.com/automl/DEHB.
+
+    For more information on the algorithm,
+    see original paper at https://arxiv.org/abs/2105.09821.
+
+    Awad, Noor, Neeratyoy Mallik, and Frank Hutter. "Dehb: Evolutionary hyperband for scalable,
+    robust and efficient hyperparameter optimization." arXiv preprint arXiv:2105.09821 (2021).
+
+    Parameters
+    ----------
+    space: `orion.algo.space.Space`
+        Optimisation space with priors for each dimension.
+
+    seed: None, int or sequence of int
+        Seed for the random number generator used to sample new trials.
+        Default: ``None``
+
+    mutation_factor: float
+        Mutation probability
+        Default: ``0.5``
+
+    crossover_prob: float
+        Crossover probability
+        Default: ``0.5``
+
+    mutation_strategy: str
+        Mutation strategy rand1, rand2dir randtobest1 currenttobest1 best1 best2 rand2
+        Default: ``'rand1'``
+
+    crossover_strategy: str
+        Crossover strategy bin or exp
+        Default: ``'bin'``
+
+    boundary_fix_type: str
+        Boundary fix method, clip or random
+        Default: ``'random'``
+
+    min_clip: float
+        Min clip when boundary fix method is clip
+        Default: ``None``
+
+    max_clip: float
+        Max clip when boundary fix method is clip
+        Default: ``None``
+
+    max_age: Optional[int]
+        Default: ``np.inf``
+
+    """
+
+    requires_type = None
+    requires_dist = None
+    requires_shape = "flattened"
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        space: Space = None,
+        seed: Optional[int] = None,
+        mutation_factor: float = 0.5,
+        crossover_prob: float = 0.5,
+        mutation_strategy: str = "rand1",
+        crossover_strategy: str = "bin",
+        boundary_fix_type: str = "random",
+        min_clip: Optional[int] = None,
+        max_clip: Optional[int] = None,
+        max_age: Optional[int] = np.inf,
```

Comment:
Also, nitpick, and totally optional, but I personally prefer using the newer, more intuitive syntax for types, like using `int | float | None`, and such.
To enable that, use `from __future__ import annotations`. Then you can also use `list[int]` instead of `List[int]`, `dict[str, Trial]` instead of `Dict[str, Trial]`, etc.

Reply:
Looks better indeed! I addressed these comments in ebfff61c98c062202355f2ce3c308661e9fc80a1. I removed `max_age` from the arguments because I was unable to find out what it is used for, it is not documented in source code nor mentioned in the paper.